src/pyobo/sources/gtdb.py

```diff
@@ -0,0 +1,111 @@
+"""Convert GTDB taxonomy to OBO format."""
+
+import logging
+from collections.abc import Iterable
+
+import pandas as pd
+from tqdm.auto import tqdm
+
+from pyobo.struct import Obo, Reference, Term
+from pyobo.struct.typedef import has_taxonomy_rank
+from pyobo.utils.path import ensure_path
+
+__all__ = [
+    "GTDBGetter",
+]
+
+PREFIX = "gtdb"
+LEVEL_ORDER = ["d__", "p__", "c__", "o__", "f__", "g__", "s__"]
+
+#: A mapping from GTDB prefixes to TAXRANK ranks
+LEVEL_TO_TAXRANK = {
+    "d__": Reference(prefix="TAXRANK", identifier="0000037", name="domain"),
+    "p__": Reference(prefix="TAXRANK", identifier="0000001", name="phylum"),
+    "c__": Reference(prefix="TAXRANK", identifier="0000002", name="class"),
+    "o__": Reference(prefix="TAXRANK", identifier="0000003", name="order"),
+    "f__": Reference(prefix="TAXRANK", identifier="0000004", name="family"),
+    "g__": Reference(prefix="TAXRANK", identifier="0000005", name="genus"),
+    "s__": Reference(prefix="TAXRANK", identifier="0000006", name="species"),
+}
+
+GTDB_AR_URL = "https://data.gtdb.ecogenomic.org/releases/latest/ar53_metadata.tsv.gz"
```

Comment:
Can you add comments on each of these URLs explaining what they are? At minimum, what is AR vs VBAC

Reply:
AR is the data set for Archaea.
BAC is the data set for Bacteria.
the goal of the resulting ontology is to combine taxonomy for archaea and bacteria.
These are the GTDB metadata files. Thee are .tsvs with over 50 columns with all sorts of information about each species (all sorts of sequence quality metrics, mapping to silva taxonomy, etc), currently only parsing two columns, one with the GTDB taxonomy string and the one with the ncbitaxon id mapping. It is important to note that there is 1 to many relantionship between GTDB to NCBI, at the extreme you will see the term "s__Staphylococcus aureus" has 4202 different ncbitax ids annotated as dbxref.